modules/web/src/com/haulmont/cuba/web/gui/components/datagrid/HierarchicalDataGridDataProvider.java

```diff
@@ -61,4 +62,20 @@ public int getChildCount(HierarchicalQuery<T, SerializablePredicate<T>> query) {
     public boolean hasChildren(T item) {
         return getTreeDataGridSource().hasChildren(item);
     }
+
+    @Override
+    public int getLevel(final T item) {
+        TreeDataGridItems<T> items = getTreeDataGridSource();
+        if (!items.containsItem(item)) {
+            throw new IllegalStateException("TreeDataGridItems do not contain the item passed to the method");
```

Comment:
1. Throw `IllegalArgumentException`
2. The caller does not know (and should not know) the internal implementation of a particular data provider, so rewrite the exception message as shown below:
`Data provider doesn't contain the item passed to the method`